CMake/FindMKL.cmake

```diff
@@ -1,43 +1,90 @@
 # Simple file to find MKL (if availible)
 # This needs a lot of work to make it robust
 INCLUDE( CheckCXXSourceCompiles )
+CMAKE_MINIMUM_REQUIRED(VERSION 3.2.0)
```

Comment:
What features of cmake need version 3.2.0?   We've been supporting back as far as 2.8.10, at the cost of some cmake version if-tests.